File: frontend/src/models/Purchase.ts
export default interface Purchase {
  id: string | null;
  fkConsoleId: string | null;
  fkGameId: string | null;
  fkPlaceOfPurchase: string;
  fkImageId: string | null;
  description: string | null;
  costBase: number;
  costTax: number;
  costShipping: number;
  constOther: number;
  costTotal: number | null;
  date: string | null;
  notes: string | null;
}

File: backend/src/main/java/org/fews/backend/model/Purchase.java
package org.fews.backend.model;

import lombok.Builder;
import lombok.Data;

import java.time.Instant;
import java.util.UUID;

@Data
@Builder
public class Purchase {
    private UUID id;
    private String name;
    private String purchaseDate;
    private double costBase;
    private double costTax;
    private double costShipping;
    private double costOther;
    private double costTotal;
    private String purchaseFrom;
    private String notes;
    private UUID imageId;
    private UUID consoleId;
    private UUID gameId;
    private Instant createdTimestamp;
    private Instant modifiedTimestamp;
}
